storage/sql.go

```diff
@@ -0,0 +1,378 @@
+package storage
```

Comment:
Should we put this under a `sql` package?

Reply:
Then it would conflict with `database/sql` everywhere. I don't mind it here, but what do you think?

Reply:
Most of the signatures contained herein are specific to `database/sql` (row iterators etc..). It's ok if `sql` conflicts with `database/sql`. I see that a lot in plenty of idiomatic Go projects. Most people end up aliasing their imports. However, if we want to avoid the collision maybe a better package name would just be `common`. So `storage/common` is a package with shared/common code across various datastores (of which the row iterators etc.. can reside).